### PR TITLE
Use onecore umbrella lib in onecore builds

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: The windowsapp.lib umbrella was being used in non-appcontainer builds. The reasoning behind it was some bug in the onecore apiset umbrellas that exists/used to exist and caused either a BVT validation failure, or a downlevel break (I wasn't the one who authored the changes and don't know the details). This workaround used to work because the windowsapp.lib umbrella is similar to the onecore apiset umbrella (both link to apisets and expose about the same set of functions).

After the latest Windows SDK update, it seems like linking to windowsapp.lib is exposing the appcontainer implementation of the delayload wrappers, and causing delayloading to fail. I'm removing the old workaround and linking to the correct umbrella libs if the build is for onecore.

This change probably deserves a downlevel run to make sure it's working. I'm waiting on the nuget pipelines (the GPU pool takes a bit) and will take a manual look into the imported symbols as soon as possible.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -1273,7 +1273,7 @@ else()
 endif()
 
 if (WIN32)
-  if (onnxruntime_BUILD_FOR_WINDOWS_STORE)
+  if (WINDOWS_STORE)
```
Comment (ryanlai2) [inline cmake/CMakeLists.txt:1276]: What is the main purpose of checking for WINDOWS_STORE instead of onnxruntime_BUILD_FOR_WINDOWS_STORE?

Comment (tiagoshibata) [reply cmake/CMakeLists.txt:1276]: Previously WINDOWS_STORE would be enabled in store and non-store to always link to windowsapp.lib, and onnxruntime_BUILD_FOR_WINDOWS_STORE was our own thing to differentiate whether we really want store or just the runtime workaround. I'm removing the workaround altogether and keeping only WINDOWS_STORE

File: cmake/target_delayload.cmake [hunk 2]
```diff
@@ -7,9 +7,9 @@ function(target_delayload target_name)
         message(SEND_ERROR "Delayloading is only supported in MSVC")
     endif()
     foreach(lib ${ARGN})
-        target_link_options(${target_name} PRIVATE /DELAYLOAD:"${lib}")
+        target_link_options(${target_name} PRIVATE /DELAYLOAD:"${lib}" ${DELAYLOAD_IMPLEMENTATION})
```
Comment (ryanlai2) [inline cmake/target_delayload.cmake:10]: How does ${DELAYLOAD_IMPLEMENTATION} get resolved? I don't see it declared anywhere

Comment (tiagoshibata) [reply cmake/target_delayload.cmake]: Yeah, just saw it and removed it :)

File: cmake/store_toolchain.cmake [hunk 3]
```diff
@@ -3,6 +3,3 @@ set(CMAKE_SYSTEM_VERSION 10.0)
 if (NOT DEFINED CMAKE_SYSTEM_PROCESSOR)
   set(CMAKE_SYSTEM_PROCESSOR ${CMAKE_HOST_SYSTEM_PROCESSOR})
 endif()
-if (NOT onnxruntime_BUILD_FOR_WINDOWS_STORE)
-  set(CMAKE_VS_GLOBALS "AppContainerApplication=false")
```
Comment (ryanlai2) [inline cmake/store_toolchain.cmake:7]: Does VS_GLOBAL_AppContainerApplication VS_GLOBAL_AppContainerApplication  and AppContainerApplication not need to be set to true / false anymore?

Comment (tiagoshibata) [reply cmake/store_toolchain.cmake:7]: VS should set it when WINDOWS_STORE, we don't have to change it now

**Review by ryanlai2 — DISMISSED** [review]

**Review by ryanlai2 — APPROVED** [review]

**Pull request merged**